payjoin/src/receive/multi_party/mod.rs

```diff
@@ -0,0 +1,268 @@
+    }
+}
+
+pub struct OutputsUnknown {
+    v1: v1::OutputsUnknown,
+    sender_contexts: Vec<SessionContext>,
+}
+
+impl OutputsUnknown {
+    pub fn identify_receiver_outputs(
+        self,
+        is_receiver_output: impl Fn(&bitcoin::Script) -> Result<bool, Error>,
+    ) -> Result<WantsOutputs, Error> {
+        let inner = self.v1.identify_receiver_outputs(is_receiver_output)?;
+        Ok(WantsOutputs { v1: inner, sender_contexts: self.sender_contexts })
+    }
+}
+
+pub struct WantsOutputs {
+    v1: v1::WantsOutputs,
+    sender_contexts: Vec<SessionContext>,
+}
+
+impl WantsOutputs {
+    pub fn commit_outputs(self) -> WantsInputs {
+        let inner = self.v1.commit_outputs();
+        WantsInputs { v1: inner, sender_contexts: self.sender_contexts }
+    }
+}
+
+pub struct WantsInputs {
+    v1: v1::WantsInputs,
+    sender_contexts: Vec<SessionContext>,
+}
+
+impl WantsInputs {
+    /// Add the provided list of inputs to the transaction.
+    /// Any excess input amount is added to the change_vout output indicated previously.
+    pub fn contribute_inputs(
+        self,
+        inputs: impl IntoIterator<Item = InputPair>,
+    ) -> Result<WantsInputs, InputContributionError> {
+        let inner = self.v1.contribute_inputs(inputs)?;
+        Ok(WantsInputs { v1: inner, sender_contexts: self.sender_contexts })
+    }
+
+    /// Proceed to the proposal finalization step.
+    /// Inputs cannot be modified after this function is called.
+    pub fn commit_inputs(self) -> ProvisionalProposal {
+        let inner = self.v1.commit_inputs();
+        ProvisionalProposal { v1: inner, sender_contexts: self.sender_contexts }
+    }
+}
+
+pub struct ProvisionalProposal {
+    v1: v1::ProvisionalProposal,
+    sender_contexts: Vec<SessionContext>,
+}
+
+impl ProvisionalProposal {
+    pub fn finalize_proposal(
+        self,
+        wallet_process_psbt: impl Fn(&Psbt) -> Result<Psbt, Error>,
+        min_feerate_sat_per_vb: Option<FeeRate>,
+        max_feerate_sat_per_vb: FeeRate,
+    ) -> Result<PayjoinProposal, Error> {
+        let inner = self.v1.finalize_proposal(
+            wallet_process_psbt,
+            min_feerate_sat_per_vb,
+            Some(max_feerate_sat_per_vb),
+        )?;
+        Ok(PayjoinProposal { v1: inner, sender_contexts: self.sender_contexts })
+    }
+}
+
+pub struct PayjoinProposal {
+    v1: v1::PayjoinProposal,
+    sender_contexts: Vec<SessionContext>,
+}
+
+impl PayjoinProposal {
+    pub fn sender_iter(&self) -> impl Iterator<Item = v2::PayjoinProposal> {
+        self.sender_contexts
+            .iter()
+            .map(|ctx| v2::PayjoinProposal::new(self.v1.clone(), ctx.clone()))
+            .collect::<Vec<_>>()
+            .into_iter()
+    }
+
+    pub fn proposal(&self) -> &v1::PayjoinProposal { &self.v1 }
+}
+
+pub struct FinalizedProposal {
+    v2_proposals: Vec<v2::UncheckedProposal>,
+}
+
+impl Default for FinalizedProposal {
+    fn default() -> Self { Self::new() }
+}
```

Comment:
I don't think this is used.

Reply:
Its not but clippy yelled at me. I'll revisit

Reply:
Are you using a client clippy config with extra rules?

Reply:
No default ones. Ussually additional clippy rules are defined in a toml at the root of the repo